### ISSUE TITLE
The AMBuildScript will mangle the path to itself and fails to configure

### ISSUE THREAD

**Issue by cademtz** [alliedmodders/hl2sdk-manifests#7]
My project uses the following file tree:
```
- configure.py
- deps
  - metamod-source
    - hl2sdk-manifests
```
*(`deps/metamod-source` is a git submodule pointing to `metamod-source`, which points to `hl2sdk-manifests`)*

Upon executing this line, the path is mangled at some point and configuration fails.
`hl2sdk_manifests` is a valid absolute path in this case:
```py
builder.Eval(os.path.join(hl2sdk_manifests, 'SdkHelpers.ambuild')
```

The resulting errors show incorrect paths:
```
python3 ../configure.py --hl2sdk-manifests /home/user/dev/recorder/deps/metamod-source/hl2sdk-manifests/
...
Configure failed: [Errno 2] No such file or directory: '/home/user/dev/recorder/home/user/dev/recorder/deps/metamod-source/hl2sdk-mani/SdkHelpers.ambuild'

python3 ../configure.py --hl2sdk-manifests ../deps/metamod-source/hl2sdk-manifests/
...
Configure failed: [Errno 2] No such file or directory: '/home/user/dev/deps/metamod-source/hl2sdk-manifests/SdkHelpers.ambuild'
```

Here is the full output:
```
Traceback (most recent call last):
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/prep.py", line 156, in Configure
    if not cm.generate(options.generator):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/context_manager.py", line 93, in generate
    self.parseBuildScripts()
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/context_manager.py", line 50, in parseBuildScripts
    self.execContext(cx)
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/context_manager.py", line 148, in execContext
    exec(code, scriptGlobals)
  File "/home/user/dev/recorder/AMBuildScript", line 85, in <module>
    SdkHelpers = builder.Eval(os.path.join(hl2sdk_manifests, 'SdkHelpers.ambuild'), {
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/context.py", line 115, in Eval
    return self.cm.evalScript(self, path, vars or {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/context_manager.py", line 60, in evalScript
    obj = self.importScriptImpl(context, path, vars or {})
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/v2_2/context_manager.py", line 77, in importScriptImpl
    self.generator.addConfigureFile(parent, scriptPath)
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/frontend/amb2_gen.py", line 744, in addConfigureFile
    self.db.add_or_update_script(path)
  File "/home/user/dev/.venv-ambuild/lib/python3.12/site-packages/ambuild2/database.py", line 826, in add_or_update_script
    stamp = os.path.getmtime(path)
            ^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen genericpath>", line 67, in getmtime
FileNotFoundError: [Errno 2] No such file or directory: '/home/user/dev/deps/metamod-source/hl2sdk-manifests/SdkHelpers.ambuild'
Configure failed: [Errno 2] No such file or directory: '/home/user/dev/deps/metamod-source/hl2sdk-manifests/SdkHelpers.ambuild'
```

**Comment by asherkin** [comment]
It's looking like the path is unconditionally treated as relative to the project root. What does this output:

```
python3 ../configure.py --hl2sdk-manifests deps/metamod-source/hl2sdk-manifests/
```

**Comment by cademtz** [comment]
Still fails:
```
python3 ../configure.py --hl2sdk-manifests deps/metamod-source/hl2sdk-manifests/
...
Configure failed: [Errno 2] No such file or directory: '/home/user/dev/recorder/hl2sdk-manifests/manifests'
```

**Comment by psychonic** [comment]
I applied a series of fix-ups to address multiple issues with path handling here.

SdkHelpers no longer requires the hl2sdk-manfiests directory to be alongside the build source. Additionally, better relative path support has now been added both within SdkHelpers for SDK paths, as well as within the Source 2 sample plugin's AMBuild script for the manifests repo and MM:S repo paths.

Note that at least one of these fixes requires the very latest AMBuild version at the time of this post.
